src/agent/src/mount.rs

```diff
@@ -288,6 +293,52 @@ async fn ephemeral_storage_handler(
     Ok("".to_string())
 }
 
+#[instrument]
+async fn nydus_storage_handler(
+    _logger: &Logger,
+    storage: &Storage,
+    sandbox: Arc<Mutex<Sandbox>>,
+) -> Result<String> {
+    let logger = _logger.new(o!("subsystem" => "mount", "driver"=> "nydus"));
+    let opts_vec: Vec<String> = storage.driver_options.to_vec();
+    let opts = parse_options(opts_vec);
+
+    let snapshot_dir=  opts.get(SNAPSHOTDIR).ok_or_else(||
+        anyhow!("failed to get snapshot_dir"))?;
+    let lower_dir = opts.get(LOWERDIR).ok_or_else(||
+        anyhow!("failed to get lower_dir"))?;
+
+
+    let mut upper_path = PathBuf::from(snapshot_dir);
```

Comment:
suggest: `upper_path` and `work_path` use the same process, can use some common code like an anonymous fn to reduce the duplicated code?